### PR TITLE
normalize deployment of loadbalancer

### PR DESCRIPTION
**Pull request by deitch**
This is a pretty fundamental change.

Currently, we do the following with CCM inside the code:

1. enable BGP appropriately
1. read the metallb manifests from within the CCM container
1. deploy those manifests to kubernetes
1. modify the metallb configmap to update BGP peers and services. This is done on initialization and every add/delete of Kubernetes node or service

We have the option to disable the above using an env var and/or a setting in the secret.

This lead to several issues:

* complex deployment management, therefore complex code
* lots of `ClusterRole` settings for CCM just so it can pass those down to metallb
* undeploying CCM does not undeploy metalllb
* one of the biggest values in terms of loadbalancers is the last point: CCM does a great job managing the metallb configmap. Customers who want to deploy their own metallb but take advantage of CCM config management have no option; it is either fully on (CCM managed all the way) or disabled (no CCM benefit for load balancers)

This PR switches that by doing the following:

* CCM does not deploy metallb. However, along with each ccm `deployment.yaml` in Releases, it also provides a `loadbalancer.yaml` to deploy the load balancer. One can apply that, or not, as preferred
* When undeploying CCM, one also can undeploy metallb (if applied that way) by undeploying the `loadbalancer.yaml`
* CCM just manages the configmap, if it finds one. If not, it just ignores it.

It further can be configured to:

* use the default configmap `metallb-system:config`
* override to a different configmap using an env var and/or secret property
* disable entirely. This is useful if you are using the default `metallb-system:config`, but are managing it on your own and want CCM out of the way

This is intended directly to address #81 along with several other issues.

This needs more testing (always), but is ready for initial reviews.

Start with the README, as it describes the new and simpler flow.

### CODE REVIEW

**Comment by deitch** [comment]
Fixes #81

**Comment by displague** [comment]
It looks like `system:cloud-controller-manager` is missing create/update/patch roles on `service` and `endpoints`, needed by https://github.com/packethost/packet-ccm/blob/metallb-manifest/packet/eip_controlplane_reconciliation.go#L201-L319

File: README.md [hunk 1]
```diff
@@ -173,34 +202,46 @@ For the control plane nodes, the Packet CCM uses static Elastic IP assignment, v
...
-  * Service deletion: find the `/32` IP address on the service spec and remove it; remove from the `ConfigMap`
-* Start an independent loop that checks every 30 seconds (configurable) for IP address reservations that are tagged with `usage="packet-ccm-auto"` and `service="<service-hash>"` but not on any services. If it finds one:
-  * If a service exists that matches the `<service-hash>`, that is an indication that an IP address reservation request was made, not completed at request time, and now is available. Add the IP to the [service spec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#servicespec-v1-core) and ensure is in the pools of the metallb `ConfigMap` with `auto-assign: false`
-  * If no service exists that is missing an IP, or none with a matching hash, delete the IP reservation
+By default, CCM controls the loadbalancer by updating the `ConfigMap` named `metall-system:config`,
```
Comment (displague) [inline README.md:205]: `metallb-system` (missing b)

Comment (deitch) [reply README.md:205]: Nice catch. Updated, thanks.

**Comment by deitch** [comment]
> It looks like system:cloud-controller-manager is missing create/update/patch roles on service and endpoints

Nice catch. Fixed up.

**Review by displague — APPROVED** [review]
This worked well for me. 

I deployed it using the cluster-api-provider-packet and ran into a snag because I deployed a single node cluster (the default configuration in the https://cluster-api.sigs.k8s.io/user/quick-start.html#initialize-the-management-cluster) and this single node included an untollerated `node-role.kubernetes.io/master` taint. Once I updated the metallb-controller to tolerate this, the sample service (and default/kubernetes) were both assigned EIPs and the services behind those IPs were accessible.

**Pull request merged**